lib/posix/options/rtsig.c

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024, Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "posix_internal.h"
+
+#include <errno.h>
+#include <time.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/posix/signal.h>
+
+int sigqueue(pid_t pid, int signo, union sigval value)
+{
+	struct posix_thread *pth = to_posix_thread(pid);
+	union k_sig_val val = {
+		.sival_ptr = value.sival_ptr,
+	};
+
+	if (pth == NULL) {
+		errno = ESRCH;
+		return -1;
+	}
+
+	return k_sig_queue(&pth->thread, signo, val);
+}
+
+int sigtimedwait(const sigset_t *ZRESTRICT set, siginfo_t *ZRESTRICT info,
+		 const struct timespec *ZRESTRICT timeout)
+{
+	int ret;
+	k_timeout_t to;
+
+	if (timeout == NULL) {
+		to = K_FOREVER;
+	} else if (timeout->tv_sec == 0 && timeout->tv_nsec == 0) {
```

Comment:
Do you really need this condition ? Isn't `K_NO_WAIT == K_MSEC(0)`  ?

Reply:
Yeah, this could be removed.

Reply:
Fixed!